### PR TITLE
LevelDB implementation added

### PR DESCRIPTION
**Pull request by berkantay**
This pull request aims to add LevelDB implementation into storage.

Thanks for the review 😃

### CODE REVIEW

File: leveldb/leveldb.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+		return nil
+	}
+
+	cache := CacheType{Data: []byte(val), Created: time.Now().Unix(), Expires: 0}
+
+	if exp > 0 {
+		cache.Expires = cache.Created + int64(exp.Seconds())
+	}
+
+	json_string, err := json.Marshal(cache)
```
Comment (efectn) [inline leveldb/leveldb.go:91]: jsonString would be better

File: leveldb/leveldb_test.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+)
+
+var testStore = New(Config{
+	"test.db",
+	nil,
+	nil,
+	nil,
+})
+
+func Test_Badger_Set(t *testing.T) {
```
Comment (efectn) [inline leveldb/leveldb_test.go:17]: Don't forget changing test names

**Review by efectn — COMMENTED** [review]
You should also add workflows (dependabot, tests etc.)

**Comment by gaby** [comment]
@berkantay I'm pretty sure that https://github.com/syndtr/goleveldb is abandoned. There's hasnt been any updates in almost a year

**Comment by gaby** [comment]
Even the official repo of LevelDB says it's in maintenance mode.

**Comment by berkantay** [comment]
@gaby It seems so

From leveldb offical repository README.md:
This repository is receiving very limited maintenance. We will only review the following types of changes.

Fixes for critical bugs, such as data loss or memory corruption
Changes absolutely needed by internally supported leveldb clients. These typically fix breakage introduced by a language/standard library/OS update.

In this case how should we proceed?

**Comment by gaby** [comment]
@berkantay That's for the c++ implementation. The Golang one hasnt seen updates sincd July 2022. If you can find a go implementation/bindings getting updates then yes, else no.

**Comment by berkantay** [comment]
It seems like all of the available go leveldb project development stopped somehow. I will close this PR and try to contribute for another database.

**Pull request closed without merging**